src/app/components/htmlMessage.tsx

```diff
@@ -9,8 +9,8 @@ type Type = <T extends any>(messageKey: string, Component: ComponentType<HTMLAtt
   ComponentType<Props & HTMLAttributes<T>>;
 
 // tslint:disable-next-line:variable-name
-const htmlMessage: Type = (messageKey, Component) => ({values, ...props}) =>
-  <FormattedHTMLMessage id={messageKey} values={values ? values : {}}>
+const htmlMessage: Type = (messageKey, Component) => ({values, id, ...props}) =>
+  <FormattedHTMLMessage id={id || messageKey} values={values ? values : {}}>
```

Comment:
I'm not sure about this. You can pass `id` because the component `htmlMessage` returns is allowed to get any html attributes which i think implies that those attributes (including `id`) would get forwarded to the dom node. 
In this case it's true for all attributes except `id`